airflow-core/src/airflow/ui/src/layouts/Details/Graph/Graph.tsx

```diff
@@ -71,18 +73,32 @@ export const Graph = () => {
     "gray.800",
   ]);
 
-  const { openGroupIds } = useOpenGroups();
+  const { openGroupIds, setAllGroupIds } = useOpenGroups();
 
   const [dependencies] = useLocalStorage<"all" | "immediate" | "tasks">(`dependencies-${dagId}`, "tasks");
   const [direction] = useLocalStorage<Direction>(`direction-${dagId}`, "RIGHT");
 
   const selectedColor = colorMode === "dark" ? selectedDarkColor : selectedLightColor;
+  const { data: graphData = { edges: [], nodes: [] } } = useStructureServiceStructureData(
+    {
+      dagId,
+      externalDependencies: dependencies === "immediate",
+      versionNumber: selectedVersion,
+    },
+    undefined,
+    { enabled: selectedVersion !== undefined },
+  );
 
-  const { data: graphData = { edges: [], nodes: [] } } = useStructureServiceStructureData({
-    dagId,
-    externalDependencies: dependencies === "immediate",
-    versionNumber: selectedVersion,
-  });
+  const { allGroupIds: observedGroupIds } = useMemo(
+    () => flattenGraphNodes(graphData.nodes),
+    [graphData.nodes],
+  );
+
+  useEffect(() => {
+    if (observedGroupIds.length) {
+      setAllGroupIds(observedGroupIds);
+    }
```

Comment:
Hmm. I am wondering if this is best approach.

The structure service does depend on a versionNumber. A user could remove all task groups between versions. But we're also not checking if the allGroupIds needs to be updated.

Reply:
i'm unclear what you mean.  if `graphData.nodes` changes, then observedGroupIds will be updated; and if that happens, then we'll update allGroupIds via `setAllGroupIds`. where's the gap?

Reply:
Let's check that allGroupIds != observedGroupIds. That would be a better if statement

Reply:
ok

Reply:
done